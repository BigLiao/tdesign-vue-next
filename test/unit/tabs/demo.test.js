import { mount } from '@vue/test-utils';

// unit test for component in examples.
describe('Tabs', () => {
  it('base demo works fine', () => {
    const demo = require('@/examples/tabs/demos/base.vue');
    const wrapper = mount(demo);
    expect(wrapper.element).toMatchSnapshot();
  });
  it('ban demo works fine', () => {
    const demo = require('@/examples/tabs/demos/ban.vue');
    const wrapper = mount(demo);
    expect(wrapper.element).toMatchSnapshot();
  });
  it('edit demo works fine', () => {
    const demo = require('@/examples/tabs/demos/edit.vue');
    const wrapper = mount(demo);
    expect(wrapper.element).toMatchSnapshot();
  });
  it('position demo works fine', () => {
    const demo = require('@/examples/tabs/demos/position.vue');
    const wrapper = mount(demo);
    expect(wrapper.element).toMatchSnapshot();
  });
  it('scroll demo works fine', () => {
    const demo = require('@/examples/tabs/demos/scroll.vue');
    const wrapper = mount(demo);
    expect(wrapper.element).toMatchSnapshot();
  });
  it('size demo works fine', () => {
    const demo = require('@/examples/tabs/demos/size.vue');
    const wrapper = mount(demo);
    expect(wrapper.element).toMatchSnapshot();
  });
  it('type demo works fine', () => {
    const demo = require('@/examples/tabs/demos/type.vue');
    const wrapper = mount(demo);
    expect(wrapper.element).toMatchSnapshot();
  });
});
