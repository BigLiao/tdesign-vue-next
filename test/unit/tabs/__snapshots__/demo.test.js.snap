// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs ban demo works fine 1`] = `
<div
  class="t-demo-tabs"
>
  <div
    class="t-demo-tabs__desc"
  >
    <p>
      目前支持常规，卡片两种类型选项卡
    </p>
     
    <button
      class="t-button t-size-m t-button--variant-outline t-button--theme-default"
      type="button"
    >
      切换到卡片型选项卡
    </button>
  </div>
   
  <div
    class="t-tabs"
  >
    <div
      class="t-tabs__header t-is-top"
    >
      <div
        class="t-tabs__nav"
      >
        <!---->
         
        <div
          class="t-tabs__nav-container t-is-top"
        >
          <!---->
           
          <!---->
           
          <div
            class="t-tabs__nav-scroll"
          >
            <div
              class="t-tabs__nav-wrap"
            >
              <div
                class="t-tabs__bar t-is-top"
              />
               
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="t-tabs__content"
    >
      <div
        class="t-tab-panel"
      >
        <p
          style="padding: 25px;"
        >
          normal选项卡1内容
        </p>
      </div>
       
      <!---->
       
      <!---->
    </div>
  </div>
</div>
`;

exports[`Tabs base demo works fine 1`] = `
<div
  class="t-demo-tabs"
>
  <p
    class="t-demo-tabs__desc"
  >
    基本选项卡
  </p>
   
  <div
    class="t-tabs"
  >
    <div
      class="t-tabs__header t-is-top"
    >
      <div
        class="t-tabs__nav"
      >
        <!---->
         
        <div
          class="t-tabs__nav-container t-is-top"
        >
          <!---->
           
          <!---->
           
          <div
            class="t-tabs__nav-scroll"
          >
            <div
              class="t-tabs__nav-wrap"
            >
              <div
                class="t-tabs__bar t-is-top"
              />
               
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="t-tabs__content"
    >
      <div
        class="t-tab-panel"
      >
        <p
          style="padding: 25px;"
        >
          选项卡1
        </p>
      </div>
       
      <!---->
       
      <!---->
    </div>
  </div>
</div>
`;

exports[`Tabs combination demo works fine 1`] = `
<div
  class="t-demo-tabs"
>
  <div
    class="t-demo-tabs__desc"
  >
    <p>
      目前支持常规，卡片两种类型选项卡
    </p>
     
    <button
      class="t-button t-size-m t-button--variant-outline t-button--theme-default"
      type="button"
    >
      切换到卡片型选项卡
    </button>
  </div>
   
  <div
    class="t-tabs"
  >
    <div
      class="t-tabs__header t-is-top"
    >
      <div
        class="t-tabs__nav"
      >
        <!---->
         
        <div
          class="t-tabs__nav-container t-is-top"
        >
          <!---->
           
          <!---->
           
          <div
            class="t-tabs__nav-scroll"
          >
            <div
              class="t-tabs__nav-wrap"
            >
              <div
                class="t-tabs__bar t-is-top"
              />
               
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="t-tabs__content"
    >
      <div
        class="t-tab-panel"
      >
        <p
          style="padding: 25px;"
        >
          选项卡1
        </p>
      </div>
       
      <!---->
       
      <!---->
       
      <!---->
       
      <!---->
       
      <!---->
       
      <!---->
       
      <!---->
       
      <!---->
       
      <!---->
    </div>
  </div>
</div>
`;

exports[`Tabs custom demo works fine 1`] = `
<div
  class="t-demo-tabs"
>
  <div
    class="t-demo-tabs__desc"
  >
    <p>
      可以动态增加选项卡，仅支持卡片型
    </p>
  </div>
   
  <div
    class="t-tabs"
  >
    <div
      class="t-tabs__header t-is-top"
    >
      <div
        class="t-tabs__nav"
      >
        <span
          class="t-tabs__add-btn t-size-m"
        >
          <svg
            class="t-icon t-icon-add"
            version="1.1"
            viewBox="0 0 1024 1024"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M544 160L544 480 864 480 864 544 544 544 544 864 480 864 480 544 160 544 160 480 480 480 480 160 544 160Z"
              fill-rule="evenodd"
            />
          </svg>
        </span>
         
        <div
          class="t-tabs__nav-container t-tabs__nav--card t-is-top t-is-addable"
        >
          <!---->
           
          <!---->
           
          <div
            class="t-tabs__nav-scroll"
          >
            <div
              class="t-tabs__nav-wrap"
            >
              <!---->
               
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="t-tabs__content"
    >
      <div
        class="t-tab-panel"
      >
        <p
          style="padding: 25px;"
        >
          原有选项卡内容
        </p>
      </div>
      <!---->
    </div>
  </div>
</div>
`;

exports[`Tabs operation demo works fine 1`] = `
<div
  class="t-demo-tabs"
>
  <div
    class="t-demo-tabs__desc"
  >
    <p>
      可以动态增加选项卡，仅支持卡片型
    </p>
  </div>
   
  <div
    class="t-tabs"
  >
    <div
      class="t-tabs__header t-is-top"
    >
      <div
        class="t-tabs__nav"
      >
        <span
          class="t-tabs__add-btn t-size-m"
        >
          <svg
            class="t-icon t-icon-add"
            version="1.1"
            viewBox="0 0 1024 1024"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M544 160L544 480 864 480 864 544 544 544 544 864 480 864 480 544 160 544 160 480 480 480 480 160 544 160Z"
              fill-rule="evenodd"
            />
          </svg>
        </span>
         
        <div
          class="t-tabs__nav-container t-tabs__nav--card t-is-top t-is-addable"
        >
          <!---->
           
          <!---->
           
          <div
            class="t-tabs__nav-scroll"
          >
            <div
              class="t-tabs__nav-wrap"
            >
              <!---->
               
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="t-tabs__content"
    >
      <div
        class="t-tab-panel"
      >
        <p
          style="padding: 25px;"
        >
          原有选项卡内容
        </p>
      </div>
      <!---->
    </div>
  </div>
</div>
`;

exports[`Tabs position demo works fine 1`] = `
<div
  class="t-demo-tabs"
>
  <div
    class="t-demo-tabs__desc"
  >
    <p>
      目前支持上下左右四个方向，但仅支持常规型选项卡
    </p>
     
    <button
      class="t-demo-tabs__btn t-button t-size-m t-button--variant-outline t-button--theme-default"
      type="button"
    >
      
      top
    
    </button>
     
    <button
      class="t-demo-tabs__btn t-button t-size-m t-button--variant-outline t-button--theme-default"
      type="button"
    >
      
      right
    
    </button>
     
    <button
      class="t-demo-tabs__btn t-button t-size-m t-button--variant-outline t-button--theme-default"
      type="button"
    >
      
      bottom
    
    </button>
     
    <button
      class="t-demo-tabs__btn t-button t-size-m t-button--variant-outline t-button--theme-default"
      type="button"
    >
      
      left
    
    </button>
  </div>
   
  <div
    class="t-tabs"
  >
    <div
      class="t-tabs__header t-is-top"
    >
      <div
        class="t-tabs__nav"
      >
        <!---->
         
        <div
          class="t-tabs__nav-container t-is-top"
        >
          <!---->
           
          <!---->
           
          <div
            class="t-tabs__nav-scroll"
          >
            <div
              class="t-tabs__nav-wrap"
            >
              <div
                class="t-tabs__bar t-is-top"
              />
               
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="t-tabs__content"
    >
      <div
        class="t-tab-panel"
      >
        <p
          style="padding: 25px;"
        >
          选项卡1
        </p>
      </div>
       
      <!---->
       
      <!---->
    </div>
  </div>
</div>
`;

exports[`Tabs size demo works fine 1`] = `
<div
  class="t-demo-tabs"
>
  <div
    class="t-demo-tabs__desc"
  >
    <p>
      目前支持medium，large两种类型选项卡
    </p>
     
    <button
      class="t-demo-tabs__btn t-button t-size-m t-button--variant-outline t-button--theme-default"
      type="button"
    >
      
      medium
    
    </button>
     
    <button
      class="t-demo-tabs__btn t-button t-size-m t-button--variant-outline t-button--theme-default"
      type="button"
    >
      
      large
    
    </button>
  </div>
   
  <div
    class="t-tabs"
  >
    <div
      class="t-tabs__header t-is-top"
    >
      <div
        class="t-tabs__nav"
      >
        <!---->
         
        <div
          class="t-tabs__nav-container t-is-top"
        >
          <!---->
           
          <!---->
           
          <div
            class="t-tabs__nav-scroll"
          >
            <div
              class="t-tabs__nav-wrap"
            >
              <div
                class="t-tabs__bar t-is-top"
              />
               
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="t-tabs__content"
    >
      <div
        class="t-tab-panel"
      >
        <p
          style="padding: 25px;"
        >
          选项卡1
        </p>
      </div>
       
      <!---->
       
      <!---->
    </div>
  </div>
   
  <br />
   
  <div
    class="t-tabs"
  >
    <div
      class="t-tabs__header t-is-top"
    >
      <div
        class="t-tabs__nav"
      >
        <!---->
         
        <div
          class="t-tabs__nav-container t-tabs__nav--card t-is-top"
        >
          <!---->
           
          <!---->
           
          <div
            class="t-tabs__nav-scroll"
          >
            <div
              class="t-tabs__nav-wrap"
            >
              <!---->
               
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="t-tabs__content"
    >
      <div
        class="t-tab-panel"
      >
        <p
          style="padding: 25px;"
        >
          选项卡1
        </p>
      </div>
       
      <!---->
       
      <!---->
    </div>
  </div>
</div>
`;
